tests/echo-socket-activated/echo-socket-activated.c

```diff
@@ -0,0 +1,156 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#define BUFFER_SIZE 1024
+#define SD_LISTEN_FDS_START 3
+
+/* This function is adapted from systemd so we don't have to link to libsystemd
```

Comment:
No reason really. I just happened to start in C so I could use directly sd_listen_fds from libsystemd.

I can rewrite it in Go with not much effort.